kube/secret.go

```diff
@@ -50,9 +60,27 @@ func MakeSecrets(secrets model.CVMap, defaults map[string]string) ([]*apiv1.Secr
 	total := 0 // Accumulated size of the values stored in 'currentSecret'
 
 	for key, value := range secrets {
-		ok, strValue := configValue(value, defaults)
-		if !ok {
-			return nil, nil, fmt.Errorf("Secret '%s' has no value", key)
+		var strValue string
+		if createHelmChart {
+			switch {
+			case value.Generator == nil || value.Generator.Type != model.GeneratorTypePassword:
```

Comment:
Right, I just meant that we could have changed `default:` to cover the `value.Generator.Type != model.GeneratorTypePassword` case instead, to make the code more readable (and maybe have a custom message).  This works too, though.